api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -110,8 +110,10 @@ type AzureManagedControlPlaneSpec struct {
 	OutboundType *ManagedControlPlaneOutboundType `json:"outboundType,omitempty"`
 
 	// SSHPublicKey is a string literal containing an ssh public key base64 encoded.
+	// Use empty string to autogenerate new key. Use null value to not set key.
 	// Immutable.
-	SSHPublicKey string `json:"sshPublicKey"`
+	// +optional
```

Comment:
This is technically a breaking change, though I'll defer to @CecileRobertMichon and @nojnhuh and others to comment on the practical cost of evolving from "required" to "optional" technically this evolution won't break existing user templates.

Reply:
I'm not too worried about the `required` -> `optional` change (although this is technically breaking for round-trip conversion) for the same reasons @nojnhuh mentions above, however, are we concerned about the type change (from `string` to `*string`)? Not sure if that would cause upgrade issues. @maciaszczykm have you been able to validate upgrade?

Reply:
@CecileRobertMichon I haven't done image upgrade on running CAPZ, but the tests that you have started have passed.

Reply:
I think w.r.t. the JSON API, the `string` -> `*string` change is backwards compatible since specifying an empty ssh key behaves the same way and not specifying one at all wasn't possible before. This is a breaking change for consumers of the Go types though, but I'm not sure how strict we've been with that in the past.

Reply:
Sounds good